plugin/src/main/kotlin/org/openrewrite/gradle/AndroidProjectSpec.kt

```diff
@@ -0,0 +1,292 @@
+        }
+
+        if (rewriteYaml != null) {
+            Files.write(dir.resolve("rewrite.yml"), rewriteYaml!!.toByteArray())
+        }
+
+        if (checkstyleXml != null) {
+            dir.resolve("config/checkstyle/checkstyle.xml").apply {
+                Files.createDirectories(parent)
+                Files.write(dir, checkstyleXml!!.toByteArray())
+            }
+        }
+
+        for (props in propertiesFiles.entries) {
+            dir.resolve(props.key).apply {
+                Files.createDirectories(parent)
+                Files.write(dir, props.value.toByteArray())
+            }
+        }
+
+        for (text in textFiles.entries) {
+            dir.resolve(text.key).apply {
+                Files.createDirectories(parent)
+                Files.write(dir, text.value.toByteArray())
+            }
+        }
+
+        for (sourceSet in sourceSets) {
+            sourceSet.build(dir.resolve("src"))
+        }
+        for (subproject in subprojects) {
+            subproject.build()
+        }
+        return this
+    }
+}
+
+class AndroidSourceSetSpec(
+    private val name: String,
+    private val sourceCharset: Charset = StandardCharsets.UTF_8,
+    private val resourceCharset: Charset = StandardCharsets.UTF_8
+) {
+    private val javaSources: MutableList<String> = mutableListOf()
+    fun java(@Language("java") source: String) {
+        javaSources.add(source.trimIndent())
+    }
+
+    private val kotlinSources: MutableList<String> = mutableListOf()
+    fun kotlin(@Language("kotlin") source: String) {
+        kotlinSources.add(source.trimIndent())
+    }
+
+    private val propertiesFiles: MutableMap<String, String> = mutableMapOf()
+    fun propertiesFile(name: String, @Language("properties") text: String) {
+        propertiesFiles[name] = text
+    }
+
+    private val yamlFiles: MutableMap<String, String> = mutableMapOf()
+    fun yamlFile(name: String, @Language("yaml") text: String) {
+        yamlFiles[name] = text
+    }
+
+    private val groovyClasses: MutableList<String> = mutableListOf()
+    fun groovyClass(@Language("groovy") source: String) {
+        groovyClasses.add(source.trimIndent())
+    }
+
+
+    @Suppress("RegExpSimplifiable")
+    fun build(dir: Path): AndroidSourceSetSpec {
+        Files.createDirectories(dir)
+        for (javaSource in javaSources) {
+            val packageDecl = if (javaSource.startsWith("package")) {
+                "package\\s+([a-zA-Z0-9.]+);".toRegex(RegexOption.MULTILINE)
+                    .find(javaSource)!!
+                    .groupValues[1]
+            } else {
+                ""
+            }.replace(".", "/")
+            val clazz = ".*(class|interface|enum)\\s+([a-zA-Z0-9-_]+)".toRegex(RegexOption.MULTILINE)
+                .find(javaSource)!!.groupValues[2]
+            val path = if (packageDecl.isEmpty()) {
+                "$name/java/$clazz.java"
+            } else {
+                "$name/java/$packageDecl/$clazz.java"
```

Comment:
You will probably want to replace the dots with slashes in the package name.

Reply:
I've deleted this class.